File: frontend/src/types/WorkOrder.ts
export interface WorkOrder {
  workOrderNo: string;
  machineNo: string;
  operatorName: string;
  orderQty: number;
  completedQty: number;
  remainingQty?: number;
}

export interface ApiResponse<T> {
  data: T | null;
  error: string | null;
  loading: boolean;
}

File: Backend/Models/WorkOrder.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    public class WorkOrder
    {
        [Key]
        public required string WorkOrderNo { get; set; }
        public required string MachineNo { get; set; }
        public required string OperatorName { get; set; }
        public int OrderQty { get; set; }
        public int CompletedQty { get; set; }
        
        [NotMapped]
        public int RemainingQty => OrderQty - CompletedQty;
    }
} 